From dc4e4e6ff28d9a7a3acd3e09298406f2d7a45eee Mon Sep 17 00:00:00 2001
From: Emmanuele Bassi <ebassi@gnome.org>
Date: Tue, 20 Jun 2017 19:39:08 +0100
Subject: [PATCH] Allow running jhbuild as root

In some environments, like a Docker container, it's perfectly acceptable
to run Jhbuild as root; in some case, it's actually required in order to
work at all without hacking around the file permissions.

For this reason, we should modify the UID check performed at startup,
and if there is a specific variable in the environment, keep running
even as root.

https://bugzilla.gnome.org/show_bug.cgi?id=783965
---
 jhbuild/main.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/jhbuild/main.py b/jhbuild/main.py
index a5cf99b54..dd722592b 100644
--- a/jhbuild/main.py
+++ b/jhbuild/main.py
@@ -94,7 +94,7 @@ def main(args):
         localedir = None
     gettext.install('jhbuild', localedir=localedir, unicode=True)
 
-    if hasattr(os, 'getuid') and os.getuid() == 0:
+    if not 'JHBUILD_RUN_AS_ROOT' in os.environ and hasattr(os, 'getuid') and os.getuid() == 0:
         sys.stderr.write(_('You should not run jhbuild as root.\n').encode(_encoding, 'replace'))
         sys.exit(1)
